src/test/java/com/nvidia/spark/rapids/jni/CastStringsTest.java

```diff
@@ -193,18 +193,71 @@ void castToDecimalNoStripTest() {
     }
   }
 
-  private void convTestInternal(Table input, Table expected, int fromBase) {
+  private void baseConversionHelper(Table input, Table expected, int fromBase,
+    boolean dropLeadingZeros) {
     try(
       ColumnVector intCol = CastStrings.toIntegersWithBase(input.getColumn(0), fromBase, false,
         DType.UINT64);
-      ColumnVector decStrCol = CastStrings.fromIntegersWithBase(intCol, 10);
-      ColumnVector hexStrCol = CastStrings.fromIntegersWithBase(intCol, 16);
+      ColumnVector decStrCol = CastStrings.fromIntegersWithBase(intCol, 10, dropLeadingZeros);
+      ColumnVector hexStrCol = CastStrings.fromIntegersWithBase(intCol, 16, dropLeadingZeros);
     ) {
       AssertUtils.assertColumnsAreEqual(expected.getColumn(0), decStrCol, "decStrCol");
       AssertUtils.assertColumnsAreEqual(expected.getColumn(1), hexStrCol, "hexStrCol");
     }
   }
 
+  private void convTestInternal(Table input, Table expected, int fromBase) {
+    baseConversionHelper(input, expected, fromBase, true);
+  }
+
+  @Test
+  void baseDec2HexTestDropLeadingZeros() {
+    try (
+        Table input = new Table.TestBuilder().column(
+            "14",
+            "2621",
+            "50"
+        ).build();
+
+        Table expected = new Table.TestBuilder().column(
+            "14",
+            "2621",
+            "50"
+        ).column(
+            "E",
+            "A3D",
+            "32"
+        ).build()
+    )
+    {
+      baseConversionHelper(input, expected, 10, true);
```

Comment:
Should we consider also testing with a different value for `fromBase`? Say, `16`?

Reply:
Testing that will not test the change and there is already an existing test for it [here](https://github.com/NVIDIA/spark-rapids-jni/blob/15b62685d1d965976644436507c54d4928652de6/src/test/java/com/nvidia/spark/rapids/jni/CastStringsTest.java#L328)

Does that work?

Reply:
Ah, yes.  Sorry I missed that.